./esbmc-wrapper.py -p ../sv-benchmarks/c/properties/unreach-call.prp -s kinduction --arch 32 ../sv-benchmarks/c/ldv-regression/fo_test.i


--------------------------------------------------------------------------------


Verifying with ESBMC
Command: ./esbmc --no-div-by-zero-check --force-malloc-success --state-hashing --no-align-check --k-step 2 --floatbv --unlimited-k-steps ../sv-benchmarks/c/ldv-regression/fo_test.i --32 --witness-output fo_test.i.graphml --no-pointer-check --no-bounds-check --interval-analysis --k-induction --max-inductive-step 3 
ESBMC version 6.8.0 64-bit x86_64 linux
Target: 32-bit little-endian x86_64-unknown-linux with esbmc libc
file ../sv-benchmarks/c/ldv-regression/fo_test.i: Parsing
Converting
Generating GOTO Program
GOTO program creation time: 0.167s
GOTO program processing time: 0.002s
*** Checking base case, k = 1
Starting Bounded Model Checking
Not unwinding 
Symex completed in: 0.003s (49 assignments)
Slicing time: 0.000s (removed 36 assignments)
Generated 1 VCC(s), 1 remaining after simplification (13 assignments)
No solver specified; defaulting to Boolector
Encoding remaining VCC(s) using bit-vector/floating-point arithmetic
Encoding to solver time: 0.000s
Solving with solver Boolector 3.2.1
Encoding to solver time: 0.000s
Runtime decision procedure: 0.003s
Building error trace

Counterexample:

State 3 file fo_test.i line 1002 function l_open thread 0
----------------------------------------------------
  fd = -2147483647 (10000000 00000000 00000000 00000001)

State 4 file fo_test.i line 3 function reach_error thread 0
----------------------------------------------------
Violated property:
  file fo_test.i line 3 function reach_error
  assertion 0
  0


VERIFICATION FAILED

Bug found (k = 1)


FALSE_REACH
